server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2210,8 +2210,12 @@ private SnapshotsInProgress updatedSnapshotsInProgress(ClusterState currentState
                                         updatedAssignmentsBuilder.put(shardId, updated);
                                     }
                                 }
-                                snapshotEntries.add(entry.withStartedShards(updatedAssignmentsBuilder.build()));
+                                final SnapshotsInProgress.Entry updatedEntry = entry.withShardStates(updatedAssignmentsBuilder.build());
+                                snapshotEntries.add(updatedEntry);
                                 changed = true;
+                                if (updatedEntry.state().completed()) {
```

Comment:
sorry if this is a stupid remark but should we do the same thing for the cloning path?

Reply:
No worries :) The clone path isn't affected by this, the logic there is different and always sets the snapshot overall state correctly. Also, there cannot be a concurrent removal of shards for clones anyway :)

Reply:
🤦🏻 makes sense, thanks!